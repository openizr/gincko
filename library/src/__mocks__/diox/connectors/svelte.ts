/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { readable } from 'svelte/store/index';

/**
 * diox/connectors/svelte mock.
 */
const connect = vi.fn(() => {
  const state = {
    steps: [{ id: 'start' }, { id: 'end' }],
    loading: process.env.LOADING === 'true',
  };
  return vi.fn(() => readable(state, (set) => {
    set(state);
  }));
});

export default connect;
