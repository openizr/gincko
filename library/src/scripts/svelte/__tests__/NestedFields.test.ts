/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import { render, waitFor } from '@testing-library/svelte';
import NestedFields from 'scripts/svelte/NestedFields.svelte';

vi.mock('biuty/svelte');
vi.mock('scripts/svelte/Field.svelte');

describe('svelte/NestedFields', () => {
  const onUserAction = vi.fn();
  const userInputs = { test: 'ok' };
  const variables = { var: 'value' };
  const i18n = vi.fn(() => 'TRANSLATED LABEL');

  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly - array', async () => {
    const { container } = render(NestedFields, {
      props: {
        t: i18n,
        type: 'array',
        label: 'Test',
        helper: 'Helper',
        path: 'path.0.to.field',
        variables,
        userInputs,
        fields: [{
          id: 'Message',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }, null],
        onUserAction,
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - array with minItems', async () => {
    const { container, rerender } = render(NestedFields, {
      props: {
        t: i18n,
        type: 'array',
        minItems: 2,
        path: 'path.0.to.field',
        variables,
        userInputs,
        fields: [{
          id: 'Message1',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }, {
          id: 'Message2',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }, {
          id: 'Message3',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }],
        onUserAction,
      },
    });
    await waitFor(() => new Promise<void>((resolve) => { setTimeout(resolve, 50); }));
    expect(container.firstChild).toMatchSnapshot();
    rerender({
      t: i18n,
      variables,
      userInputs,
      type: 'array',
      onUserAction,
      path: 'path.0.to.field',
      minItems: 4,
      fields: [{
        id: 'Message1',
        status: 'initial',
        component: 'Message',
        componentProps: {},
      }, {
        id: 'Message2',
        status: 'initial',
        component: 'Message',
        componentProps: {},
      }, null],
    });
    expect(container.firstChild).toMatchSnapshot();
    rerender({
      t: i18n,
      variables,
      userInputs,
      type: 'array',
      onUserAction,
      path: 'path.0.to.field',
      minItems: undefined,
      fields: [{
        id: 'Message1',
        status: 'initial',
        component: 'Message',
        componentProps: {},
      }, {
        id: 'Message2',
        status: 'initial',
        component: 'Message',
        componentProps: {},
      }, null],
    });
  });

  test('renders correctly - array with maxItems', async () => {
    const { container, rerender } = render(NestedFields, {
      props: {
        t: i18n,
        type: 'array',
        path: 'path.0.to.field',
        variables,
        userInputs,
        maxItems: 1,
        value: ['test', 'test'],
        fields: [{
          id: 'Message',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }, null],
        onUserAction,
      },
    });
    await waitFor(() => new Promise<void>((resolve) => { setTimeout(resolve, 50); }));
    expect(container.firstChild).toMatchSnapshot();
    rerender({
      t: i18n,
      variables,
      userInputs,
      type: 'array',
      onUserAction,
      path: 'path.0.to.field',
      maxItems: 0,
      fields: [{
        id: 'Message',
        status: 'initial',
        component: 'Message',
        componentProps: {},
      }, null],
    });
    expect(container.firstChild).toMatchSnapshot();
    rerender({
      t: i18n,
      variables,
      userInputs,
      type: 'array',
      onUserAction,
      path: 'path.0.to.field',
      maxItems: undefined,
      fields: [{
        id: 'Message',
        status: 'initial',
        component: 'Message',
        componentProps: {},
      }, null],
    });
  });

  test('renders correctly - object', async () => {
    const { container } = render(NestedFields, {
      props: {
        t: i18n,
        type: 'object',
        path: 'path.0.to.field',
        variables,
        userInputs,
        fields: [{
          id: 'Message',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }, null],
        onUserAction,
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - dynamicObject with valid pattern', async () => {
    const { container } = render(NestedFields, {
      props: {
        t: i18n,
        type: 'dynamicObject',
        path: 'path.0.to.field',
        variables,
        userInputs,
        fields: [{
          id: 'Message',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }, null],
        onUserAction,
        allowedPatterns: [/test/],
        addTextfieldProps: { debounceTimeout: 10 },
      },
    });
    await waitFor(() => new Promise<void>((resolve) => { setTimeout(resolve, 50); }));
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - dynamicObject with invalid pattern', async () => {
    const { container } = render(NestedFields, {
      props: {
        t: i18n,
        type: 'dynamicObject',
        path: 'path.0.to.field',
        variables,
        userInputs,
        fields: [{
          id: 'Message',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }, null],
        onUserAction,
        addTextfieldProps: { debounceTimeout: 10, helper: 'INVALID' },
      },
    });
    await waitFor(() => new Promise<void>((resolve) => { setTimeout(resolve, 50); }));
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - dynamicObject with error', async () => {
    const { container, rerender } = render(NestedFields, {
      props: {
        t: i18n,
        helper: 'Error',
        modifiers: 'error',
        type: 'dynamicObject',
        variables,
        path: 'path.0.to.field',
        userInputs,
        fields: [{
          id: 'Message1',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }, {
          id: 'Message2',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }, null],
        onUserAction,
      },
    });
    rerender({
      t: i18n,
      variables,
      userInputs,
      onUserAction,
      type: 'dynamicObject',
      path: 'path.0.to.field',
      value: {
        test: 'Test',
        test2: 'Test',
      },
      fields: [{
        id: 'Message1',
        status: 'initial',
        component: 'Message',
        componentProps: {},
      }, {
        id: 'Message2',
        status: 'initial',
        component: 'Message',
        componentProps: {},
      }, null],
    });
    await waitFor(() => new Promise<void>((resolve) => { setTimeout(resolve, 50); }));
    expect(container.firstChild).toMatchSnapshot();
  });
});
