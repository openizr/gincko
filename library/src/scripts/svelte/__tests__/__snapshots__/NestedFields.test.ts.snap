// Vitest Snapshot v1

exports[`svelte/NestedFields > renders correctly - array 1`] = `
<div>
  <div
    class="ui-nested-fields  array"
  >
    <span
      class="ui-nested-fields__label"
    >
      Test
    </span>
     
    
    <div
      class="ui-nested-fields__field"
    >
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div
        id="field"
      >
        {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    
     
    <div
      class="ui-nested-fields__add"
    >
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
         
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
    </div>
     
    <span
      class="ui-nested-fields__helper"
    >
      Helper
    </span>
  </div>
  <!--&lt;NestedFields&gt;-->
</div>
`;

exports[`svelte/NestedFields > renders correctly - array with maxItems 1`] = `
<div>
  <div
    class="ui-nested-fields  array"
  >
     
    
    <div
      class="ui-nested-fields__field"
    >
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div
        id="field"
      >
        {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    
     
    <div
      class="ui-nested-fields__add"
    >
       
    </div>
     
  </div>
  <!--&lt;NestedFields&gt;-->
</div>
`;

exports[`svelte/NestedFields > renders correctly - array with maxItems 2`] = `
<div>
  <div
    class="ui-nested-fields  array"
  >
     
    
    <div
      class="ui-nested-fields__field"
    >
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div
        id="field"
      >
        {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    
     
    <div
      class="ui-nested-fields__add"
    >
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
         
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
    </div>
     
  </div>
  <!--&lt;NestedFields&gt;-->
</div>
`;

exports[`svelte/NestedFields > renders correctly - array with minItems 1`] = `
<div>
  <div
    class="ui-nested-fields  array"
  >
     
    
    <div
      class="ui-nested-fields__field"
    >
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div
        id="field"
      >
        {"field":{"id":"Message1","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message1","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    <div
      class="ui-nested-fields__field"
    >
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div
        id="field"
      >
        {"field":{"id":"Message2","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message2","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    <div
      class="ui-nested-fields__field"
    >
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div
        id="field"
      >
        {"field":{"id":"Message3","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message3","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
     
    <div
      class="ui-nested-fields__add"
    >
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
         
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
    </div>
     
  </div>
  <!--&lt;NestedFields&gt;-->
</div>
`;

exports[`svelte/NestedFields > renders correctly - array with minItems 2`] = `
<div>
  <div
    class="ui-nested-fields  array"
  >
     
    
    <div
      class="ui-nested-fields__field"
    >
       
       
      <div
        id="field"
      >
        {"field":{"id":"Message1","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message1","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    <div
      class="ui-nested-fields__field"
    >
       
       
      <div
        id="field"
      >
        {"field":{"id":"Message2","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message2","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    
     
    <div
      class="ui-nested-fields__add"
    >
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
         
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
    </div>
     
  </div>
  <!--&lt;NestedFields&gt;-->
</div>
`;

exports[`svelte/NestedFields > renders correctly - dynamicObject with error 1`] = `
<div>
  <div
    class="ui-nested-fields  dynamicObject"
  >
     
    
    <div
      class="ui-nested-fields__field"
    >
      <span
        class="ui-nested-fields__field__label"
      >
        Message1
      </span>
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div
        id="field"
      >
        {"field":{"id":"Message1","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message1","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    <div
      class="ui-nested-fields__field"
    >
      <span
        class="ui-nested-fields__field__label"
      >
        Message2
      </span>
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div
        id="field"
      >
        {"field":{"id":"Message2","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message2","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    
     
    <div
      class="ui-nested-fields__add"
    >
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        null-add
         
        undefined
         
        
         
        undefined
         
        undefined
         
        100
         
        undefined
         
        true
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        disabled 
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
    </div>
     
  </div>
  <!--&lt;NestedFields&gt;-->
</div>
`;

exports[`svelte/NestedFields > renders correctly - dynamicObject with invalid pattern 1`] = `
<div>
  <div
    class="ui-nested-fields  dynamicObject"
  >
     
    
    <div
      class="ui-nested-fields__field"
    >
      <span
        class="ui-nested-fields__field__label"
      >
        Message
      </span>
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div
        id="field"
      >
        {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    
     
    <div
      class="ui-nested-fields__add"
    >
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        null-add
         
        undefined
         
        
         
        undefined
         
        INVALID
         
        10
         
        undefined
         
        true
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        disabled 
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
    </div>
     
  </div>
  <!--&lt;NestedFields&gt;-->
</div>
`;

exports[`svelte/NestedFields > renders correctly - dynamicObject with valid pattern 1`] = `
<div>
  <div
    class="ui-nested-fields  dynamicObject"
  >
     
    
    <div
      class="ui-nested-fields__field"
    >
      <span
        class="ui-nested-fields__field__label"
      >
        Message
      </span>
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div
        id="field"
      >
        {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    
     
    <div
      class="ui-nested-fields__add"
    >
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        null-add
         
        undefined
         
        
         
        undefined
         
        undefined
         
        10
         
        undefined
         
        true
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
       
      <div>
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        disabled 
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
         
        undefined
      </div>
      <!--&lt;GenericComponent&gt;-->
    </div>
     
  </div>
  <!--&lt;NestedFields&gt;-->
</div>
`;

exports[`svelte/NestedFields > renders correctly - object 1`] = `
<div>
  <div
    class="ui-nested-fields  object"
  >
     
    
    <div
      class="ui-nested-fields__field"
    >
       
       
      <div
        id="field"
      >
        {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"path":"path.0.to.field.Message","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
    </div>
    
    
    
     
     
  </div>
  <!--&lt;NestedFields&gt;-->
</div>
`;
